gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagementTaskStreamImpl.java

```diff
@@ -167,7 +158,7 @@ private DagTask createDagTask(DagActionStore.DagAction dagAction, MultiActiveLea
 
     switch (dagActionType) {
       case LAUNCH:
-        return new LaunchDagTask(dagAction, leaseObtainedStatus);
+        return new LaunchDagTask(dagAction, leaseObtainedStatus, dagActionStore.get());
```

Comment:
why `Optional` - when would the `DMTSImpl` operate w/o a `DagActionStore`?

Reply:
it wouldn't however we use guice to initialize and the `dagActionStore` is optionally bound in Guice. I am adding a check that will throw if `DagActionStore` is not present

Reply:
it's only used if warmStandbyMode is enabled. We've optionally bound all our classes that are protected by a config. If we're sure that we want to always enable warm standby we can later removal optionally bound case.